package/MDAnalysis/analysis/align.py

```diff
@@ -1120,6 +1138,30 @@ def __init__(
 
         logger.info("RMS-fitting on {0:d} atoms.".format(len(self.ref_atoms)))
 
+    def run(self, start=None, stop=None, step=None, verbose=None, **kwargs):
+        """
+        Run the analysis. If in_memory=True fall back to serial.
+        """
+        requested_backend = kwargs.pop("backend", None)
+
+        if getattr(self, "_in_memory", False):
+            # We are in the in_memory case: always run serial.
+            if requested_backend not in (None, "serial"):
+                warnings.warn(
+                    "The in-memory parallel trajectory usage is inefficient"
+                    "and not supported. Falling back to serial.",
+                    RuntimeWarning,
+                )
```

Comment:
I won't be in favor of a warning, and would rather explicitly raise ValueError because, well, how often do you switch off / ignore warnings?)